### PR TITLE
Validator: Build correct required peers set

### PR DESCRIPTION
**Pull request by omerfirmak**
Not tested yet, but hoping that it will help us eliminate some of the SCP traffic.

### CODE REVIEW

File: source/agora/node/Validator.d [hunk 1]
```diff
@@ -602,22 +601,19 @@ public class Validator : FullNode, API
...
+    private void buildRequiredKeys (NodeID self_id, in QuorumConfig quorum_conf) @safe
     {
-        import std.algorithm;
-
-        foreach (utxo; utxos.filter!(utxo => utxo != filter_utxo))
-        {
+        auto validators = this.getValidators();
+        quorum_conf.nodes.filter!(node_id => node_id != self_id)
+        .map!(node_id => validators[node_id].utxo)
+        .each!((utxo) {
```
Comment (hewison-chris) [inline source/agora/node/Validator.d]:
```suggestion
            .map!(node_id => validators[node_id].utxo)
            .each!((utxo) {
```

**Review by hewison-chris — APPROVED** [review]
LGTM, just alignment.

**Comment by omerfirmak** [comment]
Works fine in practice but it disturbed some tests.

**Comment by omerfirmak** [comment]
so much green

**Review by Geod24 — APPROVED** [review]

**Pull request merged**